.codeclimate.yml

```diff
@@ -0,0 +1,23 @@
+exclude_patterns:
+- "cypress/**"
+- "**/*.test.*"
+- "**/*.stories.*"
+- "**/config/**"
+- "**/fixture*"
+- "scripts/lighthouseBudget.js"
+- "scripts/lighthouseConfig.js"
+- "src/index.js"
+- "src/server/index.jsx"
+- "src/app/components/Metadata/index.jsx"
+- "src/app/containers/ATIAnalytics/atiUrl.js"
+- "src/app/containers/ATIAnalytics/amp/index.jsx"
+- "src/app/containers/Figure/index.jsx"
+- "src/app/components/ConsentBanner/index.jsx"
+- "src/app/containers/ConsentBanner/index.amp.jsx"
+- "src/app/Layouts/defaultPageWrapper.jsx"
+- "src/app/contexts/RequestContext/index.jsx"
+- "src/app/components/ErrorPage/index.jsx"
+- "src/app/lib/styledGrid/index.jsx"
+- "src/server/Document/component.jsx"
+- "src/app/containers/ErrorMain/index.jsx"
+- "src/app/containers/Navigation/index.jsx"
```

Comment:
There are already some "in-app" CC exclude patterns configured, from looking at https://docs.codeclimate.com/docs/advanced-configuration#section-configuration-formats I guess this file will be used in preference to those, is that the intention?

Reply:
Just mentioning because there are some in-app ones that I don't think are covered here, the full current set is:
```
config/
db/
dist/
features/
**/node_modules/
script/
**/spec/
**/test/
**/tests/
Tests/
**/vendor/
**/*_test.go
**/*.d.ts
cypress/{integration,support}/*.js
src/**/*.test.{jsx,js}
src/app/lib/config/**
```

Reply:
Yep - this is intended to overrule both the default exception list, and the in-app config that was set up on the CC UI.

I don't have access to the CC web UI to verify what was excluded, but my basic process for building this file was to disable the excludes and see what checks failed, and then adding those files to this exclusion list.

This should leave us back where we were, except the exclusion list is now in source control.